File: updated_code/code/python/1260-shift-2d-grid.py
class ShiftGrid:
    """
    A 2D grid of size m x n and an integer k. You need to shift the grid k times.
    In one shift operation:
    * Element at grid[i][j] moves to grid[i][j + 1].
    * Element at grid[i][n - 1] moves to grid[i + 1][0].
    * Element at grid[m - 1][n - 1] moves to grid[0][0].
    Return the 2D grid after applying shift operation k times.
    """
    def shiftGrid(self, grid: list[list[int]], k: int) -> list[list[int]]:
        """
        Shift grid k times.
        """
        m, n = len(grid), len(grid[0])
        k = k % (m * n)
        flat = [num for row in grid for num in row]
        flat = flat[-k:] + flat[:-k]
        rows = [list(x) for x in zip(*[iter(flat)] * n)]
        return rows

File: updated_code/code/java/1260-shift-2d-grid.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    /**
     * Shift the given 2D grid to the right by k times.
     * 
     * @param grid   The 2D grid to be shifted
     * @param k      The number of times to shift the grid
     * @return The shifted 2D grid
     */
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        if (grid == null || grid.length == 0) {
            return new ArrayList<>();
        }

        int m = grid.length; // number of rows
        int n = grid[0].length; // number of columns
        k = k % (m * n); // prevent overflow

        List<List<Integer>> result = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            List<Integer> row = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                row.add(grid[i][j]);
            }
            result.add(row);
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                int index = (i * n + j + k) % (m * n);
                result.get(index / n).set(index % n, grid[i][j]);
            }
        }

        return result;
    }
}